framework/test/unit/modules/interoperability/mainchain/module.spec.ts

```diff
@@ -0,0 +1,1185 @@
+import { BaseInteroperabilityModule } from '../../../../../src/modules/interoperability/base_interoperability_module';
+import {
+	HASH_LENGTH,
+	MIN_RETURN_FEE_PER_BYTE_BEDDOWS,
+	CHAIN_NAME_MAINCHAIN,
+	MAX_NUM_VALIDATORS,
+	MODULE_NAME_INTEROPERABILITY,
+	EMPTY_HASH,
+} from '../../../../../src/modules/interoperability/constants';
+import {
+	ChainStatus,
+	MainchainInteroperabilityModule,
+	ActiveValidator,
+	GenesisBlockExecuteContext,
+} from '../../../../../src';
+import {
+	ChainInfo,
+	GenesisInteroperability,
+} from '../../../../../src/modules/interoperability/types';
+import {
+	InMemoryPrefixedStateDB,
+	createGenesisBlockContext,
+	CreateGenesisBlockContextParams,
+} from '../../../../../src/testing';
+import {
+	computeValidatorsHash,
+	validNameCharset,
+} from '../../../../../src/modules/interoperability/utils';
+import { genesisInteroperabilitySchema } from '../../../../../src/modules/interoperability/schemas';
+import { TerminatedOutboxAccount } from '../../../../../src/modules/interoperability/stores/terminated_outbox';
+import { TerminatedStateAccount } from '../../../../../src/modules/interoperability/stores/terminated_state';
+
+const createInitGenesisStateContext = (
+	genesisInteroperability: GenesisInteroperability,
+	params: CreateGenesisBlockContextParams,
+): GenesisBlockExecuteContext => {
+	const encodedAsset = codec.encode(genesisInteroperabilitySchema, genesisInteroperability);
+
+	return createGenesisBlockContext({
+		...params,
+		assets: new BlockAssets([{ module: MODULE_NAME_INTEROPERABILITY, data: encodedAsset }]),
+	}).createInitGenesisStateContext();
+};
+
+describe('initGenesisState', () => {
+	const chainID = Buffer.from([0, 0, 0, 0]);
+	const mainchainID = Buffer.from([0, 0, 0, 0]);
+	const mainchainTokenID = Buffer.concat([mainchainID, Buffer.alloc(4)]);
+	let stateStore: PrefixedStateReadWriter;
+	let interopMod: BaseInteroperabilityModule;
+	let certificateThreshold = BigInt(0);
+
+	const channelData = {
+		inbox: {
+			appendPath: [Buffer.alloc(HASH_LENGTH), Buffer.alloc(HASH_LENGTH)],
+			root: utils.getRandomBytes(HASH_LENGTH),
+			size: 18,
+		},
+		outbox: {
+			appendPath: [Buffer.alloc(HASH_LENGTH), Buffer.alloc(HASH_LENGTH)],
+			root: utils.getRandomBytes(HASH_LENGTH),
+			size: 18,
+		},
+		partnerChainOutboxRoot: utils.getRandomBytes(HASH_LENGTH),
+		messageFeeTokenID: mainchainTokenID,
+		minReturnFeePerByte: MIN_RETURN_FEE_PER_BYTE_BEDDOWS,
+	};
+
+	const activeValidator = {
+		// utils.getRandomBytes(BLS_PUBLIC_KEY_LENGTH).toString('hex')
+		blsKey: Buffer.from(
+			'3c1e6f29e3434f816cd6697e56cc54bc8d80927bf65a1361b383aa338cd3f63cbf82ce801b752cb32f8ecb3f8cc16835',
+			'hex',
+		),
+		bftWeight: BigInt(10),
+	};
+
+	const activeValidators = [activeValidator];
+	const chainValidators = {
+		activeValidators,
+		certificateThreshold: BigInt(20),
+	};
+
+	const lastCertificate = {
+		height: 567467,
+		timestamp: Date.now() / 10000000000,
```

Comment:
Took me some time to figure out the purpose of this (divide timestamp by `10000000000`).

If I understand correctly, it makes the `lastCertificate` "old" comparing to other params that uses `Date.now()`. So I think we could simply use something like `Date.now() - 1000`?